### PR TITLE
Docs.ms ToC generation

### PR DESCRIPTION
**Pull request by danieljurek**
Generates ToC for Python ref docs on docs.ms

* docindex run -- https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1447859&view=logs&j=dc056dfc-c0cf-5958-c8c4-8da4f91a3739&t=5ada4d23-7845-5a6d-3a43-b1ccd956469f
* Docs CI run - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1447859&view=logs&j=dc056dfc-c0cf-5958-c8c4-8da4f91a3739&t=5ada4d23-7845-5a6d-3a43-b1ccd956469f

* eng/common change -- https://github.com/Azure/azure-sdk-tools/pull/2938

### CODE REVIEW

**Comment by danieljurek** [comment]
@weshaggard, PTAL

File: eng/scripts/docs/Docs-ToC.ps1 [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+  $functionService =  [PSCustomObject]@{
+    name = 'Functions';
+    landingPageType = 'Service';
+    children = @('azure-functions', 'azure-functions-durable')
+  }
+
+  # The network service is not onboarded into the regular Python build because
+  # it takes many hours to build.
+  $networkService = [PSCustomObject]@{
```
Comment (weshaggard) [inline eng/scripts/docs/Docs-ToC.ps1:105]: For these exceptions to the rule do you think it is easier to handle them in code or by putting them in a separate file that we just merge in? I'm not sure it which is easier but something to consider.

**Review by weshaggard — APPROVED** [review]

**Comment by danieljurek** [comment]
Overriding checkenforcer... This is blocking up on a known issue relating to the common engineering system and not this PR.

**Pull request merged**